### DISCUSSION TITLE
v9 support lower Redis versions?

### DISCUSSION THREAD

**Discussion by pjebs** [redis/go-redis#2241] [Q&A]
I did a quick test with v9 of this package.

It works with v7 of Redis as mentioned in docs.
When I tried it with v6 of Redis, it did not connect.

Do I have to use v8 of this package to use with v6?

Is there a chance that v9 of this package can support v6 and v7 of Redis?

**Comment by natenho** [comment]
Adding to the question, I'd like to understand the differences between major versions (regular, v8, v9), because I couldn't find in docs.

It would be great to have a migration description from one to another aswell.

**Comment by vmihailenco** [comment]
If you are using **Redis 6**, install go-redis/**v8**:

```shell
go get github.com/go-redis/redis/v8
```

If you are using **Redis 7**, install go-redis/**v9**:

```shell
go get github.com/go-redis/redis/v9
```

But `github.com/go-redis/redis/v9` should also work with Redis 6.

**Reply by sibadhish** [reply]
> But github.com/go-redis/redis/v9 should also work with Redis 6.

If v9 will work with Redis 6 then why do you recommend installing v8 for Redis 6..?
Also what is the lowest Redis Server Version supported by v9?

**Reply by Zh3Wang** [reply]
can i use v9 with redis5 ?

**Comment by khasanovbi** [comment]
What error do you have?

**Comment by sincerefly** [comment]
I used v9 with redis6, using stream and got 'ERR syntax error'

**Comment by WillyPratama96** [comment]
Hi, i want to ask regarding go-redis v8.7.1, which RESP version is supported for that? I look for the documentation for that version, but i don’t find it. Anyone know?
Thank you

**Comment by Zh3Wang** [comment]
can i use v9 with redis5 ?

**Comment by zhangdc007** [comment]
I use v9 with redis 5,init return err:Unbale to connect to RedisERR Unknown subcommand or wrong number of arguments for 'setinfo'. Try CLIENT HELP

**Reply by rmichnovicz-verkada** [reply]
fixed with https://github.com/redis/go-redis/pull/2915

**Comment by liuzengh** [comment]
Is there an official answer now?

> I did a quick test with v9 of this package.
> 
> It works with v7 of Redis as mentioned in docs. When I tried it with v6 of Redis, it did not connect.
> 
> Do I have to use v8 of this package to use with v6?
> 
> Is there a chance that v9 of this package can support v6 and v7 of Redis?

**Comment by ndyakov** [comment]
In the https://github.com/redis/go-redis/blob/master/README.md#supported-versions we have added a section for the supported versions and this will be present (and kept up to date) from the next release forward.
